api/handlers/service_plan.go

```diff
@@ -130,6 +132,22 @@ func (h *ServicePlan) updatePlanVisibility(r *http.Request) (*routing.Response,
 	return routing.NewResponse(http.StatusOK).WithBody(presenter.ForServicePlanVisibility(visibility, h.serverURL)), nil
 }
 
+func (h *ServicePlan) delete(r *http.Request) (*routing.Response, error) {
+	authInfo, _ := authorization.InfoFromContext(r.Context())
+	logger := logr.FromContextOrDiscard(r.Context()).WithName("handlers.service-plan.delete")
+
+	planGUID := routing.URLParam(r, "guid")
+	if _, err := h.servicePlanRepo.GetPlan(r.Context(), authInfo, planGUID); err != nil {
```

Comment:
Can we skip the getting of the plan and just go on and delete it?

Reply:
@georgethebeatle Done